resources/build/zip.inc.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+#
+# This script contains zip and unzip utilities to function for all environments.
+# unzip is available in all environments
+# zip is not available by default on Windows, but can be manually installed.
+# If zip is not available, then fall back to 7z for zipping.
+#
+
+# * `flags`        flags for the zip command
+# * `zip file`     zip file to create
+# * `files`        files to include in the zip file
+function zip_files() {
+  local FLAGS="$1"
+  shift 1
+  local ZIP_FILE="$1"
+  shift 1
+  local FILE_LIST="$@"
+
+  COMPRESS_CMD=zip
+  if ! command -v zip 2>&1 > /dev/null; then
+    # Fallback to 7z
+    if [[ -z "${SEVENZ+x}" ]]; then
+      case "${OSTYPE}" in
+        "cygwin")
+          SEVENZ="${SEVENZ_HOME}"/7z.exe
+          ;;
+        "msys")
+          SEVENZ="${SEVENZ_HOME}"/7z.exe
+          ;;
+        *)
+          SEVENZ=7z
+          ;;
+      esac
+
+      # 7z command, excluding build.sh files
+      "${SEVENZ}" ${FLAGS} ${ZIP_FILE} ${FILE_LIST} '-xr!build.sh'
+    fi
+  else  
+    # zip command, excluding build.sh files
+    ${COMPRESS_CMD} ${FLAGS} ${ZIP_FILE} ${FILE_LIST} -x \*\*/build\.sh
+  fi
```

Comment:
hmmm... I don't think the build.sh exclusion should be done at this level. We need that to be parameterized as well.

Tricky, not sure the best solution for this!

Reply:
I don't want to say 'design discussion' but perhaps you and @ermshiperete can put your heads together on a clean design? I would volunteer but I am travelling ...

Reply:
How about we let the caller write the files to exclude in a file and pass the filename as third parameter (e.g. `excludefile`) before `FILE_LIST`?

The excludefile could look like this:
```
*/build.sh
```

and in the script we could change the respective lines to:
```
"${SEVENZ}" ${FLAGS} ${ZIP_FILE} ${FILE_LIST} -x@excludefile
```
and
```
${COMPRESS_CMD} ${FLAGS} ${ZIP_FILE} ${FILE_LIST} -x @excludefile
```

Reply:
Would there always be an `EXCLUDEFILE` and just have a blank file if the user chooses not to exclude any?

ok, so the zip_files() parameter list would be:
$1 ZIP_FILE
$2 EXCLUDEFILE
rest of parameters are flags or FILE_LIST

where flags are: `a`, `-*`
and files are everything else

Reply:
k, the latest iteration has the required 2nd parameter of a filename containing list of files to exclude.
I suppose it's a blank file if there's nothing to exclude?

Reply:
blank or missing file would be best.